src/Schema/Comparator.php

```diff
@@ -275,14 +275,17 @@ public function compareTables(Table $oldTable, Table $newTable): TableDiff
             foreach ($newForeignKeys as $newKey => $newForeignKey) {
                 if ($this->diffForeignKey($oldForeignKey, $newForeignKey) === false) {
                     unset($oldForeignKeys[$oldKey], $newForeignKeys[$newKey]);
-                } else {
-                    if (strtolower($oldForeignKey->getName()) === strtolower($newForeignKey->getName())) {
-                        $droppedForeignKeys[$oldKey] = $oldForeignKey;
-                        $addedForeignKeys[$newKey]   = $newForeignKey;
+                    continue 2;
```

Comment:
As I understand, this is the fix. The rest of the lines have been mostly affected to satisfy the coding standard.

The essence of the fix is that once we have found a matching pair of new key / old key, we not only continue the inner loop, we continue the outer one as well. Otherwise, the another equal key will also match.

Is that correct?

Reply:
Yes, the `continue 2;` is the actual change. It stops comparing the new key when a matching old key has been found.